nightwatch maintenance

yanfali opened the pull request:
- some minor fixes to nightwatch bootstrapping process
- try to fix up the security warnings from github and npm

@mechmerlin can you please test this on linux? I've only tried it out on OSX. Thank you.

yanfali commented on functional_tests/selenium_setup.sh:
```diff
@@ -17,15 +14,20 @@ CHR_DRIVER_VER="2.40"
     CHR_DRIVER="https://chromedriver.storage.googleapis.com/${CHR_DRIVER_VER}/chromedriver_mac64.zip"
     GKO_DRIVER="https://github.com/mozilla/geckodriver/releases/download/v${GKO_DRIVER_VER}/geckodriver-v${GKO_DRIVER_VER}-macos.tar.gz"
     DRIVER="mac64"
+    ROOTUSER="root:staff"
```
I'll remove these. no longer needed

yanfali commented on functional_tests/selenium_setup.sh:
```diff
@@ -17,15 +14,20 @@ CHR_DRIVER_VER="2.40"
+    ROOTUSER="root:staff"
 else
     VERSION="$(uname -m)"
+    ROOTUSER="root:root"
```
ditto

mechmerlin commented:
This occurred on a fresh install of Ubuntu 18.04, while following the documentation.

```> qmk_configurator@0.1.1 nightwatch /home/merlin/Documents/qmk_configurator
> node_modules/nightwatch/bin/nightwatch -e chromeHeadless

sh: 1: node_modules/nightwatch/bin/nightwatch: not found

npm ERR! Linux 4.15.0-34-generic
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "nightwatch"
npm ERR! node v8.10.0
npm ERR! npm  v3.5.2
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! qmk_configurator@0.1.1 nightwatch: `node_modules/nightwatch/bin/nightwatch -e chromeHeadless`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the qmk_configurator@0.1.1 nightwatch script 'node_modules/nightwatch/bin/nightwatch -e chromeHeadless'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the qmk_configurator package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node_modules/nightwatch/bin/nightwatch -e chromeHeadless
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs qmk_configurator
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls qmk_configurator
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/merlin/Documents/qmk_configurator/npm-debug.log
```

yanfali commented:
This could be a node version issue. I'll look into using nvm to control what version people run

yanfali commented:
@mechmerlin ok I've updated the README me to include instructions for using nvm (node version manager so we don't have to deal with the old version on ubuntu. I've tested this on Ubuntu 18.04.1 LTS. You may have problems with chrome, you need a very recent version of chrome that supports headless operation. I added a link to an article that explains how to install that on ubuntu.

Now I get this

```console
vagrant@ubuntu1804:~/qmk_configurator$ npm run nightwatch

> qmk_configurator@0.1.1 nightwatch /home/vagrant/qmk_configurator
> node_modules/nightwatch/bin/nightwatch -e chromeHeadless


[Qmk Base Page Test] Test Suite
===============================
Running:  The backend status bar is displayed and has the required elements
```

yanfali commented:
@jackhumbert this one is good to go. Confirmed working by nim and merlin. Thank you.

jackhumbert commented:
Awesome - thanks!
